### PR TITLE
ENH: Add predict methods to NominalGEE and OrdinalGEE

### PR DESCRIPTION
**Pull request by huntermills707**
## Summary (updated)

This PR adds model-level `predict` methods to `NominalGEE` and `OrdinalGEE` that operate on the original `(nobs, k)` design matrix and return category probabilities. The inherited GLM-style `predict` methods for these classes were removed in release 0.12.0 because they returned the expanded-indicator mean rather than per-category probabilities. This implementation restores the methods with behavior appropriate for multinomial/ordinal responses, and hardens the surrounding code paths affected by overriding `predict`.

## API

```python
result.predict(exog=None, exposure=None, offset=None, which="mean", linear=None)
```

* `which="mean"` (default): returns the full `(nobs, ncat)` category-probability matrix. Columns are ordered by the sorted unique response values (`model.endog_values`). For `NominalGEE` the reference category (the largest value) is the last column.
* `which="linear"`: returns the linear predictor — one column per non-reference category for `NominalGEE`, or per threshold for `OrdinalGEE`.
* `offset` follows the GLM convention: the model offset is used when `exog` is None; a user-provided offset overrides it.
* `linear=` is accepted as a deprecated alias for `which="linear"` (emits `FutureWarning`), matching `GLM.predict`.
* `exposure=` is accepted only for signature compatibility with `GLM.predict`. It is not meaningful for these models and raises `ValueError` if provided.
* Works through the formula API: `result.predict(exog=DataFrame)` transforms the new data via the original (pre-expansion) formula specification.

## Implementation notes

* `NominalGEE.predict` uses a numerically stable log-sum-exp softmax. The stabilizing maximum includes the reference category's linear predictor (0) so that the result stays finite and normalized even when every non-reference linear predictor is large and negative.
* `OrdinalGEE.predict` computes cumulative survival probabilities via the binomial link inverse and differences them to obtain category probabilities. Because the model uses a shared slope with one intercept per threshold (proportional odds), the category probabilities are valid for any `exog` as long as the fitted thresholds are monotone. The thresholds are estimated without an ordering constraint, so in the rare degenerate case where they are not monotone a `ValueWarning` is issued and some probabilities may be negative.
* `mu` is overridden in `NominalGEEResults` and `OrdinalGEEResults` so that the expanded-indicator-scale fitted values used by residuals, deviance, etc. remain unchanged.
* `GEE._derivative_exog` is routed through a small `_ExpandedMeanModel` adapter (which returns `family.fitted(exog @ params)`) for the elasticity and dummy/count marginal-effect paths. This decouples marginal effects from the overridden `predict`, fixing `get_margeff` for `NominalGEE`/`OrdinalGEE`, and is byte-for-byte unchanged for every other GEE family.
* `NominalGEE.predict` and `OrdinalGEE.predict` validate that the `params`/`exog` dimensions are consistent and raise a clear `ValueError` on mismatch.
* `get_distribution` raises `NotImplementedError` on both models and their results classes: a single scipy frozen distribution cannot represent a  multinomial/ordinal response (previously this errored obscurely for nominal and returned a meaningless distribution for ordinal).
* `NominalGEE.__init__`/`OrdinalGEE.__init__` preserve the formula model specification (these classes expand `endog`/`exog` before calling `super().__init__`, which would otherwise drop the spec passed by `from_formula`), enabling `predict` on new data supplied as a DataFrame.

## Tests

Added/strengthened tests in `statsmodels/genmod/tests/test_gee.py`:

* `NominalGEE.predict` validated against `MNLogit` and against R's `nnet::multinom` (pinned probabilities), using `NominalGEE(Independence)`'s reduction to multinomial logit; plus a non-circular check against the legacy `_MultinomialLogit.inverse`.
* `OrdinalGEE.predict` validated by an independent `scipy.expit` + survival-differencing reconstruction, with pinned probabilitie.
* Offset handling, the deprecated `linear=` keyword, invalid `which`, `exposure` rejection, and `params`/`exog` dimension validation.
* The `NominalGEE` reference-dominant case (no NaN/overflow) and the ordinal non-monotone-threshold warning.
* `get_margeff` elasticity and dummy/count effect paths for both classes.
* The `get_distribution` guard.
* `predict` through the formula API (`from_formula`), including new data passed as a DataFrame.

All genmod tests pass:

```bash
  python -m pytest statsmodels/genmod/tests/
  # 801 passed, 2 xfailed
```

## Known limitations (pre-existing, out of scope)

These are unrelated to the `predict` math and are flagged for separate issues:

* `get_margeff(dummy=True/count=True)` with the default `at="overall"` computes delta-method standard errors via `GLM._derivative_predict`, which raises `NotImplementedError` for any model carrying an offset. `NominalGEE`/`OrdinalGEE` always carry an internal (zeros) offset from the expanded design, so the standard-error path is unavailable for them (`at="all"`, which skips SEs, works). This predates the PR.
* `NominalGEE.mean_deriv_exog` has a shape bug when the number of covariates differs from the number of non-reference categories (`k != ncut`), which breaks `get_margeff` for such models. This predates the PR; the new marginal-effects tests use `k == ncut` designs to avoid it.

## Related issues

* Relates to #7071 (predict / margins exog support).
* Relates to #7068 (ordinal intercept / identification; the threshold parameterization used by `predict` matches the existing model).
* These predict methods were previously listed as removed in the release 0.12.0 notes.

## Open question for maintainers

The contribution guidelines ask for a release-notes entry, but there is no `docs/source/release/versionX.X.rst` file for the next release yet. Should I create `version0.15.0.rst` (or another version) and add this change there, or will the release notes be handled separately?

### CODE REVIEW

**Comment by Aniketsy** [comment]
@huntermills707 thanks for the work ❤️  

> The contribution guidelines ask for a release-notes entry, but there is no docs/source/release/versionX.X.rst file for the next release yet. Should I create version0.15.0.rst (or another version) and add this change there, or will the release notes be handled separately?

Release note entry is auto generated, i don't think you need to create `0.15.0.rst `

**Comment by huntermills707** [comment]
Happy to contribute!

Open to any/all suggestions or modifications.

**Comment by huntermills707** [comment]
Realized I did not add formula support, and added this. Also added a few more tests/guards, and updated PR summary.

Between there were a few more down stream items/guards/tests/doc-strings, this PR turned into more code than I originally thought.

Happy to close this, and split into two PRs; one for Nominal GEE predict, and one for Ordered GEE predict. Assuming that is preferable.

**Comment by Aniketsy** [comment]
hmm quite large changes 😁 , initially thought came into my mind of split the changes into two PR's, i thought that would be more easy to review, maybe i could also have pointed some suggestion 😊 . But its fine

> Happy to close this, and split into two PRs; one for Nominal GEE predict, and one for Ordered GEE predict. Assuming that is preferable.

lets not close this as you have invested your time so closing this and splitting into two PR, will create extra work for you.

But yes, i think this will take time in moving forward... and getting reviewed. 

Thanks again for the improvements
